### DISCUSSION TITLE
Updating Option B: Clone the Git Repository

### DISCUSSION THREAD

**Discussion by maxzapp** [netbox-community/netbox#7621] [Q&A]
Hi,

i am at 3.0.7 and i was using this update option for the last couple of updates. But now i got this error message

@netbox:/opt/netbox$ sudo git checkout master
fatal: not a git repository (or any of the parent directories): .git
it-ops@netbox:/opt/netbox$ sudo git pull origin master
fatal: not a git repository (or any of the parent directories): .git

is there a known issue?

THX

**Comment by jeremystretch** [comment]
You appear to have deleted  the `.git/` directory from your installation path somehow. Are you sure this is where you have NetBox installed?

**Comment by maxzapp** [comment]
Hi Jeremy,

well i am pretty shure , i was not aware of such a directory:  Here is ll on the installation dir.

it-ops@netbox:/opt/netbox$ ll
total 100
drwxrwxr-x  8 root root  4096 Okt 12 20:04 ./
drwxr-xr-x  4 root root  4096 Okt 12 19:25 ../
-rw-rw-r--  1 root root  2923 Okt  8 18:04 base_requirements.txt
-rw-rw-r--  1 root root   117 Okt  8 18:04 CHANGELOG.md
drwxrwxr-x  2 root root  4096 Okt  8 18:04 contrib/
-rw-rw-r--  1 root root  9224 Okt  8 18:04 CONTRIBUTING.md
drwxrwxr-x 15 root root  4096 Okt  8 18:04 docs/
-rw-rw-r--  1 root root   222 Okt  8 18:04 .gitattributes
drwxrwxr-x  4 root root  4096 Okt  8 18:04 .github/
-rw-rw-r--  1 root root   458 Okt  8 18:04 .gitignore
-rw-r--r--  1 root root   486 Okt 12 19:38 gunicorn.py
-rw-rw-r--  1 root root 10174 Okt  8 18:04 LICENSE.txt
-rw-rw-r--  1 root root  5148 Okt  8 18:04 mkdocs.yml
drwxrwxr-x 17 root root  4096 Okt 12 20:05 netbox/
-rw-rw-r--  1 root root    72 Okt  8 18:04 NOTICE
-rw-rw-r--  1 root root  3766 Okt  8 18:04 README.md
-rw-rw-r--  1 root root   616 Okt  8 18:04 requirements.txt
drwxrwxr-x  3 root root  4096 Okt  8 18:04 scripts/
-rwxrwxr-x  1 root root  3799 Okt  8 18:04 upgrade.sh*
drwxr-xr-x  6 root root  4096 Okt 12 20:04 venv/
it-ops@netbox:/opt/netbox$

**Comment by maxzapp** [comment]
Hi,

i created a .git directory in /opt/netbox/  but same result

**Reply by maxzapp** [reply]
Ok I checked with my productive system there is .git with some content. 
Can I clone this to my 
Testsystem?

**Reply by candlerb** [reply]
It would be safe to clone the whole `/opt/netbox` tree from your production system.

Copying just `/opt/netbox/.git` is risky, because the checked-out files may not be in a state consistent with the contents of the `.git` directory.  If you did do this, then doing `git reset --hard` ought to force the rest of the tree into sync with the repo.

The question you need to ask yourself is, how did your system get into this state?  Perhaps you copied the tree using something like
```
cp -r /opt/netbox/* /some/dest/
```
which would only have copied the visible directories, and not those starting with dot.

**Reply by maxzapp** [reply]
@candlerb thx this worked. I had a backup of the netbox folder with the folder .git  so i restored it and did additionaly "git reset --hard". In the beginning i was using upgrading with option A, then it tested version B  but there was a problem  so i made some kopies maby the .git was not proper transfered (but .github was, strange)
